File: backend/Models/Shape.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models;

[Table("shapes")]
public class Shape
{
    [Column("shape_id")]
    [Required]
    public int ShapeId { get; set; }
    
    [Column("shape_pt_lat")]
    [Required]
    public float ShapePtLat { get; set; }
    
    [Column("shape_pt_lon")]
    [Required]
    public float ShapePtLon { get; set; }
    
    [Column("shape_pt_sequence")]
    [Required]
    public int ShapePtSequence { get; set; }
    
    [Column("shape_dist_traveled")]
    public float ShapeDistTraveled { get; set; }
}

File: backend/TransportRealtime/models/shape.go
package models

import "github.com/twpayne/go-geom"

type Shape struct {
	ShapeId            string      `json:"shape_id"`
	ShapePtLat         float64     `json:"shape_pt_lat"`
	ShapePtLon         float64     `json:"shape_pt_lon"`
	ShapeGeom          *geom.Point `json:"shape_geom"`
	ShapePtSequence    int         `json:"shape_pt_sequence"`
	ShapeDistTravelled float64     `json:"shape_dist_traveled"`
	Mode               string      `json:"mode"`
}
